File: 1° semestre/APC verão/Raylan da Silva Sales_lab_2/Python/Lab2a.py
import random
import math
R = list(range(0, 22))
G = list(range(0, 22))
B = list(range(0, 22))
Rf = list(range(0, 20))
Gf = list(range(0, 20))
Bf = list(range(0, 20))
Dist = list(range(0, 20))
media = 0

R.insert(0, 0)
R.insert(21, 0)
G.insert(0, 0)
G.insert(21, 0)
B.insert(0, 0)
B.insert(21, 0)
random.seed()
for i in range(1, 20):
    R[i] = random.randint(0, 255)
    G[i] = random.randint(0, 255)
    B[i] = random.randint(0, 255)
    i+1

for i in range(0,20):
    Rf[i] = (R[i]+R[i+1]+R[i+2])/3   
    Gf[i] = (G[i]+G[i+1]+G[i+2])/3
    Bf[i] = (B[i]+B[i+1]+B[i+2])/3

for i in range(0,20):
    print('Rf[{}] = {:.2f}\n'.format(i, Rf[i]))    

print('---------------')
for i in range(0,20):
    print('Gf[{}] = {:.2f}\n'.format(i, Gf[i]))    

print('---------------')
for i in range(0,20):
    print('Bf[{}] = {:.2f}\n'.format(i, Bf[i])) 

for i in range(0,20):
    Dist[i] = math.sqrt(pow((R[i]-Rf[i]),2)+ pow((G[i]-Gf[i]),2)+ pow((B[i]-Bf[i]),2))

#print('---------------')
#for i in range(0,20):
#   print('Distancia = {:.2f}'.format(Dist[i]))

for i in range(0,20):
    media += Dist[i]

media = media/20
print('---------------')
print('A distancia euclidiana media entre os 2 vetores eh: {:.2f}'.format(media))

File: 1° semestre/APC verão/Raylan da Silva Sales_lab_2/C/Lab2a.c
#include<stdio.h>
#include<stdlib.h>
#include<math.h>
#include<time.h>
int main(){
	int R[22],G[22],B[22],i,Rf[20],Gf[20],Bf[20],Dist[20];
	float media;
	R[0]=0;
	R[21]=0;
	G[0]=0;
	G[21]=0;
	B[0]=0;
	B[21]=0;
	srand(time(NULL));
	for(i=1;i<=20;i++){
		R[i] = rand() % 255;
		G[i] = rand() % 255;
		B[i] = rand() % 255;
	}
	for(i=0;i<20;i++){
		Rf[i] = (R[i]+R[i+1]+R[i+2])/3;
		Gf[i] = (G[i]+G[i+1]+G[i+2])/3;
		Bf[i] = (B[i]+B[i+1]+B[i+2])/3;
	}
	for(i=0;i<20;i++){
		printf("Rf[%d] = %d\n",i,Rf[i]);
	}
	printf("---------------\n");
	for(i=0;i<20;i++){
		printf("Gf[%d] = %d\n",i,Gf[i]);
	}
	printf("---------------\n");
	for(i=0;i<20;i++){
		printf("Bf[%d] = %d\n",i,Bf[i]);
	}
	for(i=0;i<20;i++){
		Dist[i] = sqrt(pow((R[i]-Rf[i]),2)+ pow((G[i]-Gf[i]),2)+ pow((B[i]-Bf[i]),2));
	}
	/*printf("---------------\n");
	for(i=0;i<20;i++){
		printf("Distancia = %d\n",Dist[i]);
	}*/
	for(i=0;i<20;i++){
		media +=Dist[i];
	}
	media= media/20;
	printf("---------------\n");
	printf("A distancia euclidiana media entre os 2 vetores eh: %.2f",media);
	
	return 0;
}